podpac/core/node.py

```diff
@@ -485,266 +525,160 @@ def del_cache(self, key=None, coordinates=None):
-                cPickle.dump(obj, fid)#, protocol=cPickle.HIGHEST_PROTOCOL)
+    cache_key = 'output'
+
+    @functools.wraps(fn)
+    def wrapper(self, coordinates, output=None):
+        if self.debug:
+            self._requested_coordinates = coordinates
+        key = cache_key
+        cache_coordinates = coordinates.transpose(*sorted(coordinates.dims)) # order agnostic caching
+        if self.has_cache(key, cache_coordinates) and not self.cache_update:
+            data = self.get_cache(key, cache_coordinates)
+            if output is not None:
+                order = [dim for dim in output.dims if dim not in data.dims] + list(data.dims)
+                output.transpose(*order)[:] = data
+            self._from_cache = True
         else:
-            s3 = boto3.resource('s3').Bucket(settings.S3_BUCKET_NAME)
-            io = BytesIO(cPickle.dumps(obj))
-            s3.upload_fileobj(io, path)
+            data = fn(self, coordinates, output=output,)
+            if self.cache_output:
+                self.put_cache(data, key, cache_coordinates, overwrite=self.cache_update,
+                               raise_no_cache_exception=False)
+            self._from_cache = False
 
-    def load_cached_obj(self, filename):
-        """Retreive an object from cache
+        # transpose data to match the dims order of the requested coordinates
+        order = [dim for dim in coordinates.dims if dim in data.dims]
+        data = data.transpose(*order)
 
-        Parameters
-        ----------
-        filename : str
-            File name of object to be retrieved from cache
-
-        Returns
-        -------
-        object
-            Object loaded from cache
-
-        .. deprecated:: 0.2.0
-            This method will be removed and replaced by the caching module by version 0.2.0.
-        """
-
-        import warnings
-        warnings.warn('Node.load_cached_obj will be replaced by get_cache in a later release', DeprecationWarning)
-
-        try:
-            import cPickle  # Python 2.7
-        except:
-            import _pickle as cPickle
-
-        try:
-            import boto3
-        except:
-            boto3 = None
-
-        path = self.cache_path(filename)
-        if settings.S3_BUCKET_NAME is None or not settings.CACHE_TO_S3:
-            with open(path, 'rb') as fid:
-                obj = cPickle.load(fid)
-        else:
-            s3 = boto3.resource('s3').Bucket(settings.S3_BUCKET_NAME)
-            io = BytesIO()
-            s3.download_fileobj(path, io)
-            io.seek(0)
-            obj = cPickle.loads(io.read())
-        return obj
+        if self.debug:
+            self._output = data
 
-    def clear_disk_cache(self, attr='*', node_cache=False, all_cache=False):
-        """Helper function to clear disk cache.
+        return data
+    return wrapper
 
-        WARNING: This function will permanently delete cached values
+def cached_property(key, depends=None, raise_no_cache_exception=False):
```

Comment:
I figured out how to do the dependencies on this decorator by setting up observers on traits. It complex, and the final implementation feels a little hacky, but it works. 
Turns out if you tag an attr, it will automatically become a `depends` for any cached property. This is because changing this attr changes the definition of the node, which gets used in the cache key. I think this is appropriate behaviour, but took me a while to figure out what was going on.

Reply:
Cool!! I didn't think you would get to doing this decorator.